app/safe/internal/state/io-encrypt.go

```diff
@@ -53,14 +56,34 @@ func encryptToWriterAge(out io.Writer, data string) error {
 	return nil
 }
 
+var lastEncryptToWriterAesCall time.Time
+
 func encryptToWriterAes(out io.Writer, data string) error {
+	// Calling this method too frequently can result in a less-than random IV,
+	// which can be used to break the encryption when combined with other
+	// attack vectors. Therefore, we throttle calls to this method.
+	if time.Since(lastEncryptToWriterAesCall) < time.Millisecond*time.Duration(
+		env.SafeIvInitializationInterval(),
+	) {
+		return errors.New("Calls too frequent")
+	}
+
+	lastEncryptToWriterAesCall = time.Now()
```

Comment:
This is a security-measure against time-based attacks.

If the function is called too frequently, then the initialization vector might not be “random enough” (depending on the OS), which might be a security weakness.

Thins the throttling factor will be system-dependent, I set up an env var for that and arbitrarily chose 50ms as a default value. — it can be modified on a "per need" basis.